master/pkg/container/state.go

```diff
@@ -57,3 +58,21 @@ func (s *State) UnmarshalText(text []byte) error {
 	*s = parsed
 	return nil
 }
+
+// Proto returns the proto representation of the container state.
+func (s State) Proto() containerv1.State {
+	switch s {
+	case Assigned:
+		return containerv1.State_STATE_ASSIGNED
+	case Pulling:
+		return containerv1.State_STATE_PULLING
+	case Starting:
+		return containerv1.State_STATE_STARTING
+	case Running:
+		return containerv1.State_STATE_RUNNING
+	case Terminated:
+		return containerv1.State_STATE_TERMINATED
+	default:
+		return containerv1.State_STATE_UNSPECIFIED
```

Comment:
I think things like this are better as a `panic`; we sort of talked about this and it's indicative of the program being in a bad, unrecoverable state. But I'd say leave it as is or return an `error`, since that's pretty opinionated and I think more agree on this.

Reply:
Agreed on the panic, but for now, I am leaving this as is.